infrastructure/src/test/java/org/corfudb/infrastructure/management/failuredetector/FailureDetectorTestContext.java

```diff
@@ -0,0 +1,157 @@
+package org.corfudb.infrastructure.management.failuredetector;
+
+import lombok.Getter;
+import org.corfudb.infrastructure.NodeNames;
+import org.corfudb.infrastructure.datastore.DataStore;
+import org.corfudb.infrastructure.management.ClusterAdvisor;
+import org.corfudb.infrastructure.management.CompleteGraphAdvisor;
+import org.corfudb.infrastructure.management.FileSystemAdvisor;
+import org.corfudb.infrastructure.management.PollReport;
+import org.corfudb.infrastructure.management.failuredetector.FailuresAgent.FailuresAgentBuilder;
+import org.corfudb.infrastructure.management.failuredetector.HealingAgent.HealingAgentBuilder;
+import org.corfudb.protocols.wireprotocol.ClusterState;
+import org.corfudb.protocols.wireprotocol.NodeState;
+import org.corfudb.protocols.wireprotocol.failuredetector.FileSystemStats;
+import org.corfudb.protocols.wireprotocol.failuredetector.FileSystemStats.PartitionAttributeStats;
+import org.corfudb.protocols.wireprotocol.failuredetector.NodeConnectivity;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.util.concurrent.SingletonResource;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+
+import static org.corfudb.infrastructure.management.NodeStateTestUtil.nodeState;
+import static org.corfudb.protocols.wireprotocol.failuredetector.NodeConnectivity.ConnectionStatus.OK;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+public class FailureDetectorTestContext {
+    public final String localEndpoint;
+
+    private FailureDetectorDataStore.FailureDetectorDataStoreBuilder fdDataStoreBuilder = FailureDetectorDataStore.builder();
```

Comment:
It looks like these fields can be made final.

Reply:
The context is mutable, the idea - is to have a default context, but if you want you can change any part of it in your tests, that's why fields are not mutable. The current version of the ctx, is a first version, hope we will improve it in the following pr-s